src/hotspot/share/runtime/objectMonitor.cpp

```diff
@@ -1989,30 +2007,21 @@ int ObjectMonitor::TrySpin(JavaThread* current) {
   }
 
   // Spin failed with prejudice -- reduce _SpinDuration.
-  // TODO: Use an AIMD-like policy to adjust _SpinDuration.
-  // AIMD is globally stable.
-  {
-    int x = _SpinDuration;
-    if (x > 0) {
-      // Consider an AIMD scheme like: x -= (x >> 3) + 100
-      // This is globally sample and tends to damp the response.
-      x -= Knob_Penalty;
-      if (x < 0) x = 0;
-      _SpinDuration = x;
-    }
+  if (ctr < 0) {
+    _SpinDuration = adjust_down(_SpinDuration);
   }
 
- Abort:
```

Comment:
Okay. It took a few back-and-forth pans of the code, but I've
convinced myself that your switch from `goto Abort` is equivalent
to your `break` logic *except when the break occurs when `ctr < 0` is true.*
In that case, you'll now do the `adjust_down()` call when the `goto Abort`
would have skipped it.

I don't know (yet) if this causes a problem, but it does appear to be another
change in behavior.

Reply:
Yes, the only time we adjust down is when we hit the spin count, but the code in Abort has to be run on all paths to fix up succ and fence.  Not sure if that following TryLock would ever succeed.  Seems very unlikely!